ingestion/functions/README.md

```diff
@@ -134,6 +134,22 @@ sam deploy
 From the base `ingestion/functions` dir. The deployment configuration will be
 inferred from the `samconfig.toml` file. Follow the confirmation dialogues.
 
+### Writing a parser where deduplication of patients cannot be done
+
+Some sources do not provide a unique ID for each case allowing us to update existing cases in subsequent parsing runs.
+
+To accomodate for that, here is the procedure to write a parser that only imports data that is three days old (a reasonable threshold chosen arbitrarily):
+
+NOTE: This is a proposed method and being implemented in #743.
+
+1. write the parser and have it respect the `dedupeStrategy`
```

Comment:
2 is not going to work as it would have to understand the data being parsed (the dates part of it at least), so it's already a parser and not a generic parsing_lib anymore...

1 is what I had in mind yes. Each parser get the `dedupeStrategy` params passed to them in the input event (fetched from the Source object by the retrieval function, and they have to respect it.

I am having a hard time picking up good names for those dedupe strategy params, I renamed them now to use an enum and a single number of days, hopefully it makes things a bit clearer.

Reply:
I don't think (2) would have to be "specialized" per parser. Isn't all of the data in a canonical date format in the same field (the date in the `confirmed` event)?

(1) would certainly work. It sounds like the idea is to introduce logic into the parser in order to be able to quickly change the behavior via config to deal with the initial upload process and any subsequent changes to the nature of the data?

Reply:
Ah sorry I think I see what you mean now, so the parsers would create all cases regardless of the dedupe settings and the parsing_lib would *then* drop the ones that do not match the dedupe settings before they are sent to the server, I like that approach because then there's no need for anything special in the parsers. I changed the readme to mention that approach now.

Reply:
Yep, SG!